### PR TITLE
Feat/contact support

### PR DESCRIPTION
**Pull request by ThomasWongHKG**
Plz review my code
the following content are about the Profile's Contact Support and the FAQ section

### CODE REVIEW

File: lib/screens/profile/contact_support.dart [hunk 1]
```diff
@@ -7,152 +11,468 @@ class ContactSupport extends StatefulWidget {
...
+                ),
+                DeviceSpecList(deviceInfos: deviceInfos),
+              ] else
+                FutureBuilder<String>(
+                  future: ajaxCall,
+                  builder:
+                      (BuildContext context, AsyncSnapshot<String> snapshot) {
+                    Widget returnWidget;
+                    if (snapshot.hasData) {
+                      returnWidget = SendingResultPage(
```
Comment (andreasgan) [inline lib/screens/profile/contact_support.dart:208]: instead of this variable, you can just return directly.

Comment (andreasgan) [reply lib/screens/profile/contact_support.dart:208]: for all 3

File: lib/screens/profile/contact_support.dart [hunk 2]
```diff
@@ -7,152 +11,468 @@ class ContactSupport extends StatefulWidget {
...
+                ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class textFieldInput extends StatelessWidget {
```
Comment (andreasgan) [inline lib/screens/profile/contact_support.dart:253]:
```suggestion
class _TextFieldInput extends StatelessWidget {
```

Comment (andreasgan) [reply lib/screens/profile/contact_support.dart:253]: see the other comment about _ / private classes

File: lib/screens/profile/contact_support.dart [hunk 3]
```diff
@@ -7,152 +11,468 @@ class ContactSupport extends StatefulWidget {
...
+        onChanged: (_) => textFldOnChange(),
       ),
-      // child: OptionList(languageList),
     );
   }
 }
 
-class SendButton extends StatelessWidget {
-  const SendButton({
+class DeviceSpecList extends StatelessWidget {
```
Comment (andreasgan) [inline lib/screens/profile/contact_support.dart:306]: _DeviceInfo maybe
and the inner one can be _DeviceInfoList

File: lib/screens/profile/contact_support.dart [hunk 4]
```diff
@@ -7,152 +11,468 @@ class ContactSupport extends StatefulWidget {
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Expanded(
+      child: Column(
+        children: <Widget>[
+          Expanded(
+            child: Padding(
+              padding: EdgeInsets.only(
+                  top: MediaQuery.of(context).size.height * 0.3),
```
Comment (andreasgan) [inline lib/screens/profile/contact_support.dart:420]: Why is the padding based on the window height?

Comment (ThomasWongHKG) [reply lib/screens/profile/contact_support.dart:420]: I was trying to achieve to make the msg widget to be center of the screen, dispite the appbar and the bottom and the notification bar, so I teyied to implement the width with the widget's height 
           for example: top: (MediaQuery.of(context).size.height - Widget's height) / 2

but in order to do that, I don't have sufficient skill and knowledge, plus the approaches to do that are not so easy or will used up a lot of resource (usually applies to animation which aims to re-size the layout based on the changes)  

base the things mentioned above, the end-product turns out like this :\

refs: https://stackoverflow.com/questions/49307677/how-to-get-height-of-a-widget

File: lib/screens/profile/contact_support.dart [hunk 5]
```diff
@@ -7,152 +11,461 @@ class ContactSupport extends StatefulWidget {
...
+                ),
+            ],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _textFieldInput extends StatelessWidget {
```
Comment (andreasgan) [inline lib/screens/profile/contact_support.dart:245]: capitalize the t

File: lib/screens/profile/contact_support.dart [hunk 6]
```diff
@@ -197,23 +189,23 @@ class _ContactSupportState extends State<ContactSupport> {
...
                 FutureBuilder<String>(
                   future: ajaxCall,
                   builder:
                       (BuildContext context, AsyncSnapshot<String> snapshot) {
-                    Widget returnWidget;
+                    // Widget returnWidget;
                     if (snapshot.hasData) {
-                      returnWidget = SendingResultPage(
-                        sendingResult: false,
+                      return SendingResultPage(
```
Comment (andreasgan) [inline lib/screens/profile/contact_support.dart:200]: SendingResultPage -> _SendingResultPage. Do this for all except the primary widget for the file.

File: lib/screens/profile/contact_support.dart [hunk 7]
```diff
@@ -197,23 +189,23 @@ class _ContactSupportState extends State<ContactSupport> {
...
                 ),
-                DeviceSpecList(deviceInfos: deviceInfos),
+                _DeviceInfo(deviceInfos: deviceInfos()),
               ] else
                 FutureBuilder<String>(
                   future: ajaxCall,
                   builder:
                       (BuildContext context, AsyncSnapshot<String> snapshot) {
-                    Widget returnWidget;
+                    // Widget returnWidget;
```
Comment (andreasgan) [inline lib/screens/profile/contact_support.dart:198]: better to just delete the line :)

**Review by andreasgan — CHANGES_REQUESTED** [review]

**Review by ThomasWongHKG — COMMENTED** [review]
The private access modifier have been implemented into the widgets and methods and debug print result, comment are also removed

**Pull request merged**